src/inserter.rs

```diff
@@ -264,6 +277,12 @@ where
         let result = self.insert().await;
         self.ticks.reschedule();
         result?;
+
+        if let Some(cb) = &mut self.on_commit {
+            if quantities.transactions > 0 {
+                (cb)(&quantities);
+            }
+        }
```

Comment:
You could lift this into `insert()` along with `mem::replace(&mut self.pending, Quanities::ZERO)` and the behavior should still be correct.

Reply:
addressed in [ee6b162](https://github.com/ClickHouse/clickhouse-rs/pull/307/commits/ee6b162e56a6e879c26c1f06938abea670ee1538)